src/Servers/Kestrel/Core/src/Internal/TlsConnectionFeature.cs

```diff
@@ -65,7 +66,21 @@ public SslProtocols Protocol
         set => _protocol = value;
     }
 
+    public SslStream? SslStream
+    {
+        get => _sslStream;
```

Comment:
Another part of #46442 is making sure _sslStream is consistently set by all of the UseHttps APIs. Right now only some of them set this field.

Reply:
Oops, I knew that and I blanked.

Reply:
Okay, several hops later...

`TlsConnectionFeature` always has an `SslStream`, but `TlsConnectionFeature` is only constructed if `HttpsConnectionMiddleware.OnConnectionAsync` is called.  This suggests some `UseHttps` overloads are not missing merely the `SslStream`, but all of `ITlsConnectionFeature`, `ITlsApplicationProtocolFeature`, and `ITlsHandshakeFeature`.  Or I'm missing something...

Reply:
Also, assuming my comment from my previous change is correct, even the overloads of `UseHttps` that reference `OnConnectionAsync` won't actually invoke it if only http/3 is enabled.

Reply:
Staring at the code, I can't find an overload that doesn't ultimately use `HttpsConnectionMiddleware.OnConnectionAsync` (assuming http/1 and/or http/2 is enabled).  Do you have a scenario in mind @Tratcher?

Reply:
Ah, my bad. It's not _sslStream that's null, it's that we don't consistently set the SslStream into the feature collection. 
https://github.com/dotnet/aspnetcore/blob/7d0b688e7b2241290346a98901708ea574db7f69/src/Servers/Kestrel/Core/src/Middleware/HttpsConnectionMiddleware.cs#L419
Your new feature addresses that problem. I'd suggest removing this line that adds SslStream directly to the feature collection, but that would probably break people.

Reply:
We still don't support ITlsConnectionFeature for HTTP/3? I remember asking MsQuic to expose that information, but I don't remember if we ever got it.

Reply:
Nevermind: https://github.com/dotnet/runtime/issues/70184